### PR TITLE
Very basic implementation of conditionals for uniques

### PR DESCRIPTION
**Pull request by xlenstra**
This PR implements one of the ideas brought up in #5091, optional conditionals for uniques.
It implements two of these, `<when at war>` and `<when not at war>`, and lays most of the groundwork for future implementation of other conditionals.
It implements the checking of these conditionals by adding `getMatchingApplyingUniques()` and `hasApplyingUnique()` functions, which function similar to `getMatchingUniques()` and `hasUnique()`, except that they also filter out uniques that don't apply in this situation. Currently, this can be determined only by passing the correct `civInfo` to these functions, but later on these functions will have a myriad of optional parameters that may be necessary to check in conditionals.

Although the amount of changes in this PR may look daunting, most of them are either replacing `getMatchingUniques()` with `getMatchingApplyingUniques()` or small coding style changes.

Deprecated uniques:
- [amount]% Growth [cityFilter] when not at war

Added conditionals:
- \<when at war\>
- \<when not at war\>

### CODE REVIEW

**Comment by SimonCeder** [comment]
Nice.
Perhaps the city states `allyBonus` `friendBonus` per era can be simplified to a long list of uniques with <when allies> <during the Classical Era> and so on?

**Comment by xlenstra** [comment]
> Nice.
> Perhaps the city states `allyBonus` `friendBonus` per era can be simplified to a long list of uniques with \<when allies\> \<during the Classical Era\> and so on?

I'd personally say that they should still be saved in the eras.json (unless we create a cityState.json), but they can be restated as `[stats] <from [relationFilter] City-States>` or something, so that would definitely be possible

File: core/src/com/unciv/models/translations/Translations.kt [hunk 1]
```diff
@@ -269,7 +273,16 @@ fun String.tr(): String {
...
+fun String.getPlaceholderText() = 
+    this.replace(squareBraceRegex, "[]")
+        .replace(pointyBraceRegex, "")
+        // So, this is a quick hack, but it works as long as nobody uses word separators different from " " (space) and "" (none).
+        // According to https://linguistics.stackexchange.com/questions/6131/is-there-a-long-list-of-languages-whose-writing-systems-dont-use-spaces
+        // This is a reasonable but not fully correct assumption to make.
+        // By doing it like these, we exclude languages such as Tibetan, Dzongkha (Bhutan), and Ethiopian.
+        // If we ever start getting translations for these, we'll fix them then.
+        .replace("  ", " ")
+        .trim()
```
Comment (yairm210) [inline core/src/com/unciv/models/translations/Translations.kt:285]: So this is where the magic happens...

**Comment by yairm210** [comment]
Instead of changing the function from hasMatchingUniques to hasMatchingApplyingUniques, which will require a huge rename, why not have an overload? (I assume more will follow)
getMatchingUniques(placeholderText)
getMatchingUniques(placeholderText, civInfo)

This will allow more fine-grained conditionals for specific uniques in the future, like:
getMatchingUniques(placeholderText, cityInfo)
getMatchingUniques(placeholderText, unitInfo)
etc.

**Comment by yairm210** [comment]
This is especially relevant since the need for multiple filters was raised first and foremost through the multiple unit filters (military water units etc)

**Comment by yairm210** [comment]
There's a lot of fluff here getting in the way of the actually important parts of the PR.
Please remove all renames, AND check the translation - I didn't see any changes in the .tr() function, and from what I see all translations with conditionals will be totally broken.
It seems to me that conditionals will need to go at the end of the line.

Unless ALL conditionals are set in a specific place, e.g. the end of the line, I don't see a way to handle this. You can't just "splice" an existing translation and insert conditionals in the middle and hope it parses...
And now the question becomes, "can all languages handle delimiter-conditionals". Sounds like a question for Discord.

**Comment by xlenstra** [comment]
> There's a lot of fluff here getting in the way of the actually important parts of the PR.
> Please remove all renames, AND check the translation - I didn't see any changes in the .tr() function, and from what I see all translations with conditionals will be totally broken.
> It seems to me that conditionals will need to go at the end of the line.
> 
> Unless ALL conditionals are set in a specific place, e.g. the end of the line, I don't see a way to handle this. You can't just "splice" an existing translation and insert conditionals in the middle and hope it parses...
> And now the question becomes, "can all languages handle delimiter-conditionals". Sounds like a question for Discord.

Tbh, I completely forgot about everything that has to do with translations. I'll wait for the feedback in discord and recreate this PR based on that. I'll also remove all the unnecessary fluff and start with only a few uniques being able to use conditionals, which should remove most of the renames.

**Pull request closed without merging**